### PR TITLE
Fixes #217

### LINKED ISSUE #217: Empty cells/fields are filled with values from preceding rows
I came across this issue using [dsq](https://github.com/multiprocessio/dsq) and was able to narrow is down to this project.

The issue arises with the attached file [fail.xlsx](https://github.com/multiprocessio/datastation/files/8462526/fail.xlsx) which has the following structure:

A | B | C | D
-- | -- | -- | --
a1 | b1 | c1 | d1
a2 | b2 | c2 |  
a3 | b3 |   |  
a4 |   |   |  

Dumping the file's structure using `dsq` is as follow:

```bash
> dsq --pretty fail.xlsx
+----+----+----+----+
| A  | B  | C  | D  |
+----+----+----+----+
| a1 | b1 | c1 | d1 |
| a2 | b2 | c2 | d1 |
| a3 | b3 | c2 | d1 |
| a4 | b3 | c2 | d1 |
+----+----+----+----+
```

### PR DESCRIPTION
**Pull request by steirico**
This is a fix for #217.

### CODE REVIEW

File: runner/file.go [hunk 1]
```diff
@@ -184,14 +184,14 @@ func writeSheet(rows [][]string, w *jsonutil.StreamEncoder) error {
...
 
-	row := map[string]any{}
 	for _, r := range rows {
 		if isHeader {
 			header = r
 			isHeader = false
 			continue
 		}
 
+		row := map[string]any{}
```
Comment (eatonphil) [inline runner/file.go:194]: There is a huge perf impact to this. Rather than doing this we should set the empty cells to nil each time.

**Comment by eatonphil** [comment]
Thanks for the bug report and the suggests fix! If you'd like to make the suggested change go for it otherwise I can.

File: runner/file.go (hunk 1 above)
Comment (steirico) [reply runner/file.go:194]: I'm not sure if allocating an empty map has more or less impact than iterating over `header` and setting map entries to nil.

Therefore, feel free to go for the way you prefer. 😉

**Comment by eatonphil** [comment]
I opened a PR against your branch:

https://github.com/steirico/datastation/pull/1

It also adds your reproduction file as a regression test.

**Comment by eatonphil** [comment]
These tests are failing because of your permissions as a PR author. They are ok to fail. When the rest finish I'll merge.

**Pull request merged**